### ISSUE TITLE
"omf install" Array index out of bounds error

### ISSUE THREAD

**Issue by AndreTheHunter** [fish-shell/oh-my-fish#472]
Running `omf install` throws an error:
```
test: Missing argument at index 1
Installing ... Array index out of bounds
~/.oh-my-fish/plugins/omf/omf.git.fish (line 4):       set -l path $argv[3]
                                                                         ^
in function 'omf.git'
	called on line 31 of file ~/.oh-my-fish/plugins/omf/omf.packages.install.fish
	with parameter list '--clone https://github.com/oh-my-fish/theme-'

in function 'omf.packages.install'
	called on line 28 of file ~/.oh-my-fish/plugins/omf/omf.packages.fish
	with parameter list '--theme'

in function 'omf.packages'
	called on line 11 of file ~/.oh-my-fish/plugins/omf/omf.fish
	with parameter list '--install'

in function 'omf'
	called on standard input
	with parameter list 'install'
```
Many of the other commands also produce errors.
`omf list`
```
Plugins: 
Array index out of bounds
~/.oh-my-fish/plugins/omf/omf.log.fish (line 5):       echo -e (set_color $argv[1])$argv[2..-1](set_color normal)
                                                                                        ^
in function 'omf.log'
	called on line 45 of file ~/.oh-my-fish/plugins/omf/omf.packages.fish
	with parameter list 'normal'

in function 'omf.packages'
	called on line 15 of file ~/.oh-my-fish/plugins/omf/omf.fish
	with parameter list '--list'

in function 'omf'
	called on standard input
	with parameter list 'list'


Theme: 
Array index out of bounds
~/.oh-my-fish/plugins/omf/omf.log.fish (line 5):       echo -e (set_color $argv[1])$argv[2..-1](set_color normal)
                                                                                        ^
in function 'omf.log'
	called on line 49 of file ~/.oh-my-fish/plugins/omf/omf.packages.fish
	with parameter list 'normal'

in function 'omf.packages'
	called on line 15 of file ~/.oh-my-fish/plugins/omf/omf.fish
	with parameter list '--list'

in function 'omf'
	called on standard input
	with parameter list 'list'
```
`omf update`
```
test: Missing argument at index 1
test: Missing argument at index 1
All packages were already updated
```

Setting the `config.fish` to the absolute minimum doesn't help
```
set fish_path ~/.oh-my-fish
source $fish_path/oh-my-fish.fish
```
Reinstalling omf did not help.

Running the latest oh-my-fish (pulled master from git@github.com:oh-my-fish/oh-my-fish.git) and fish version 2.2b1-20-g7efbcc0

**Comment by bpinto** [comment]
I haven't tested it with fish non-final releases, perhaps you could try using version 2.1.2? I'll try installing from git later.

**Comment by AndreTheHunter** [comment]
Changing to fish 2.1.2 didn't help (brew makes it so easy) but when running `omf list`, I got the same error:
```
Plugins: 
fish: Could not expand string '(set_color $argv[1])$argv[2..-1](set_color normal)'
~/.oh-my-fish/plugins/omf/omf.log.fish (line 15):       echo -e (set_color $argv[1])$argv[2..-1](set_color normal)
                                                                                   ^
in function 'omf.log',
  called on line 45 of file '~/.oh-my-fish/plugins/omf/omf.packages.fish',
  with parameter list 'normal'

in function 'omf.packages',
  called on line 15 of file '~/.oh-my-fish/plugins/omf/omf.fish',
  with parameter list '--list'

in function 'omf',
  called on standard input,
  with parameter list 'list'


Theme: 
fish: Could not expand string '(set_color $argv[1])$argv[2..-1](set_color normal)'
~/.oh-my-fish/plugins/omf/omf.log.fish (line 15):       echo -e (set_color $argv[1])$argv[2..-1](set_color normal)
                                                                                   ^
in function 'omf.log',
  called on line 49 of file '~/.oh-my-fish/plugins/omf/omf.packages.fish',
  with parameter list 'normal'

in function 'omf.packages',
  called on line 15 of file '~/.oh-my-fish/plugins/omf/omf.fish',
  with parameter list '--list'

in function 'omf',
  called on standard input,
  with parameter list 'list'
```

**Comment by bpinto** [comment]
May you show me your `config.fish` file?

**Comment by AndreTheHunter** [comment]
Setting my `config.fish` to the bare minimum of
```
set fish_path ~/.oh-my-fish
source $fish_path/oh-my-fish.fish
```
still throws the same error

**Comment by macdaddyaz** [comment]
I am getting the same kind of errors. Fish 2.1.1 installed via Ubuntu Vivid universe repo. Freshly installed oh-my-fish today using the "install.fish" method from the README. I found that if I switched to a different theme (I tried 'scorphish'), then I no longer get the errors.

**Comment by asyncsrc** [comment]
I had something similar happen and ended up needing to follow the manual installation steps here -- https://github.com/oh-my-fish/oh-my-fish/wiki/Manual-Installation -- which allowed me to use the most up to date set of files. 

 In my particular case, it appears that the omf.git.fish file was missing under ~/.oh-my-fish/plugins/omf/ after doing the normal/recommended install using the shell script, which prevented any plugins from being installed.

**Comment by AndreTheHunter** [comment]
I think I found the cause of the problem. In my config I was declaring `set fish_path ~/.oh-my-fish` but changing it to the template default of `set fish_path $HOME/.oh-my-fish` worked.

**Comment by AndreTheHunter** [comment]
So this issue is still occurring: 
```console
~➜ fish --version
fish, version 2.2.0
~➜ omf install
test: Missing argument at index 1
Installing ... Array index out of bounds
~/.oh-my-fish/plugins/omf/omf.git.fish (line 4):       set -l path $argv[3]
                                                                         ^
in function 'omf.git'
	called on line 31 of file ~/.oh-my-fish/plugins/omf/omf.packages.install.fish
	with parameter list '--clone https://github.com/oh-my-fish/theme-'

in function 'omf.packages.install'
	called on line 28 of file ~/.oh-my-fish/plugins/omf/omf.packages.fish
	with parameter list '--theme'

in function 'omf.packages'
	called on line 11 of file ~/.oh-my-fish/plugins/omf/omf.fish
	with parameter list '--install'

in function 'omf'
	called on standard input
	with parameter list 'install'

Username for 'https://github.com': 
```
but `omf list` works fine

**Comment by bpinto** [comment]
Fixed with #512
